src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/dagligReise/DagligReiseVedtakController.kt

```diff
@@ -0,0 +1,117 @@
+import no.nav.tilleggsstonader.sak.vedtak.dto.tilLagretVedtaksperiodeDto
+import no.nav.tilleggsstonader.sak.vedtak.forslag.ForeslåVedtaksperiodeService
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.VilkårService
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/api/vedtak/daglig-reise")
+@ProtectedWithClaims(issuer = "azuread")
+class DagligReiseVedtakController(
+    private val behandlingService: BehandlingService,
+    private val beregningService: OffentligTransportBeregningService,
+    private val tilgangService: TilgangService,
+    private val stegService: StegService,
+    private val steg: DagligReiseBeregnYtelseSteg,
+    private val vedtakService: VedtakService,
+    private val vedtakDtoMapper: VedtakDtoMapper,
+    private val foreslåVedtaksperiodeService: ForeslåVedtaksperiodeService,
+    private val vilkårService: VilkårService,
+) {
+    @PostMapping("{behandlingId}/innvilgelse")
+    fun innvilge(
+        @PathVariable behandlingId: BehandlingId,
+        @RequestBody vedtak: InnvilgelseDagligReiseRequest,
+    ) {
+        lagreVedtak(behandlingId, vedtak)
+    }
+
+    @GetMapping("{behandlingId}")
+    fun hentVedtak(
+        @PathVariable behandlingId: BehandlingId,
+    ): VedtakResponse? {
+        tilgangService.settBehandlingsdetaljerForRequest(behandlingId)
+        tilgangService.validerTilgangTilBehandling(behandlingId, AuditLoggerEvent.ACCESS)
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val vedtak = vedtakService.hentVedtak(behandlingId) ?: return null
+        return vedtakDtoMapper.toDto(vedtak, null, behandling.forrigeIverksatteBehandlingId)
+    }
+
+    @PostMapping("{behandlingId}/beregn")
+    fun beregn(
+        @PathVariable behandlingId: BehandlingId,
+        @RequestBody vedtak: InnvilgelseDagligReiseRequest,
+    ): Beregningsresultat {
+        tilgangService.settBehandlingsdetaljerForRequest(behandlingId)
+        val vilkår = vilkårService.hentVilkår(behandlingId)
+        val utgifter =
+            vilkår
+                .filter { it.offentligTransport != null }
+                .map { vilkår ->
+                    UtgiftOffentligTransport(
+                        fom = vilkår.fom!!,
+                        tom = vilkår.tom!!,
+                        antallReisedagerPerUke = vilkår.offentligTransport?.reisedagerPerUke!!,
+                        prisEnkelbillett = vilkår.offentligTransport.prisEnkelbillett,
+                        pris30dagersbillett = vilkår.offentligTransport.prisTrettidagersbillett,
+                    )
+                }
+
+        return beregningService
```

Comment:
Kan det være fint å la beregningservice håndtere innhenting av vilkår? 
Sånn er det utført for tilsyn barn og boutgifter